aws/components/ddb/state.ftl

```diff
@@ -0,0 +1,278 @@
+            [#local secretEngine = secretLink.Configuration.Solution.Engine ]
+
+            [#if secretLink.Core.Type == SECRETSTORE_COMPONENT_TYPE ]
+
+                [#local baselineLinks = getBaselineLinks(occurrence, [ "Encryption"], true, false )]
+
+                [#if baselineLinks?has_content ]
+                    [#local baselineComponentIds = getBaselineComponentIds(baselineLinks)]
+                    [#local cmkKeyId = (baselineComponentIds["Encryption"]) ]
+                [#else]
+                    [#local cmkKeyId = ""]
+                [/#if]
+
+                [#local resources = mergeObjects(
+                    resources,
+                    {
+                        "rootCredentials" :
+                            getComponentSecretResources(
+                                occurrence,
+                                "RootCredentials",
+                                "RootCredentials",
+                                cmkKeyId,
+                                secretEngine,
+                                "Root credentials for database"
+                            )
+                    })]
+
+                [#local roles = mergeObjects(
+                    roles,
+                    {
+                        "Outbound" : {
+                            "default" : "root",
+                            "root" : secretsManagerReadPermission(resources["rootCredentials"]["secret"].Id, cmkKeyId)
+                        }
+                    }
+                )]
+
+                [#local attributes = mergeObjects(
+                    attributes,
+                    {
+                        "SECRET_ARN" : getExistingReference(resources["rootCredentials"]["secret"].Id, ARN_ATTRIBUTE_TYPE)
+                    }
+                )]
+            [/#if]
+
+            [#local attributes = mergeObjects(
+                attributes,
+                {
+                    "USERNAME" : solution["rootCredential:SecretStore"].UsernameAttribute,
+                    "PASSWORD" : solution["rootCredential:SecretStore"].PasswordAttribute
+                }
+            )]
+            [#break]
+    [/#switch]
+
+    [#local resources = mergeObjects(
+        resources,
+        {
+            "dbCluster" : {
+                "Id" : id,
+                "Name" : core.FullName,
+                "Port" : port,
+                "Type" : AWS_DDS_CLUSTER_RESOURCE_TYPE,
+                "Monitored" : true
+            },
+            "dbClusterParamGroup" : {
+                "Id" : formatResourceId(AWS_DDS_CLUSTER_PARAMETER_GROUP_RESOURCE_TYPE, core.Id, replaceAlphaNumericOnly(family, "X") ),
+                "Family" : family,
+                "Type" : AWS_DDS_CLUSTER_PARAMETER_GROUP_RESOURCE_TYPE
+            }
+        }
+    )]
+
+    [#-- Calcuate the number of fixed instances required --]
+    [#if multiAZ!false ]
```

Comment:
```suggestion
    [#if solution.MultiAZ ]
```
This will also need to be added to the core id definition. We now manage multiAZ on a component level basis